### PR TITLE
Catch NAN being computed in GetDensityGlycol routine to prevent -1.#IND outputs

### PR DESCRIPTION
**Pull request by JasonGlazer**
Fix for #5139 found that a NAN was being generated by GetDensityGlycol in FluidProperties.cc.

### CODE REVIEW

**Comment by mjwitte** [comment]
Trapping that is fine (unless it's slow to check every call to fluid properties), but the real culprit is probably higher up the food chain - something may be sending NAN into the fluid properties function.

File: src/EnergyPlus/FluidProperties.cc [hunk 1]
```diff
@@ -4899,6 +4899,7 @@ namespace FluidProperties {
 		int GlycolNum;
 		bool LowErrorThisTime;
 		bool HighErrorThisTime;
+		static bool showedInterpError (false);
```
Comment (mbadams5) [inline src/EnergyPlus/FluidProperties.cc]: @JasonGlazer Please don't add new function static variables. It just makes unit testing more difficult and require effort to put it in an anonymous namespace and clear_state in the future. Just go ahead and put this new variable into an anonymous namespace and clear_state now.

**Comment by mbadams5** [comment]
I'm a little hesitant about adding a check for every call of `GetDensityGlycol`.

**Comment by Myoldmopar** [comment]
OK, so the performance suite is doing its job.  The results for this branch show a [1% performance regression](http://nrel.github.io/EnergyPlusBuildResults/EnergyPlus-4e56895873f7c1afa8910bfd782e2015922af596-x86_64-Linux-Ubuntu-14.04-gcc-4.8.html#performance).  Which, isn't much, but also doesn't keep us going in the right direction.  I'm curious about alternative ways to handle this.  I'm afraid there is some divide by zero action going on up the call chain that is driving the temperature to NaN.  If it is something more subtle, like a convergence loop that ends up forcing a temperature to infinity, then the check should probably be made higher up in the iteration loop.

**Comment by JasonGlazer** [comment]
The Microsoft Visual Studio C++ compiler has a way to stop on a NAN no matter where or when it is generated. I would expect other compilers do also. To cause it to stop on NAN the /fp:strict should be used as shown at:

http://stackoverflow.com/questions/4454582/visual-studio-c-2008-2010-break-on-float-nan

I could not get it to work by just putting it into the Visual Studio GUI and instead put it on the 

CMAKE_CXX_FLAGS 
CMAKE_C_FLAGS

variables in CMake.

In addition, I had add the fp_control_state to main.cc:

```
int
main( int argc, const char * argv[] )
{
    unsigned int fp_control_state = _controlfp( _EM_INEXACT, _MCW_EM );
    ProcessArgs( argc, argv );
    EnergyPlusPgm();
}
```

This was removed prior to checking in the branch. I am documented this here to help others in the future if they have problems with NANs that they cannot track down.

**Comment by mjwitte** [comment]
@JasonGlazer It would be good to put this in the wiki somewhere, maybe a developers tips page if we don't have a place yet.

**Comment by mjwitte** [comment]
BTW, this branch needs to be merged up to current develop.

**Comment by JasonGlazer** [comment]
I just created a new page on the wiki called [Debugging Tips](https://github.com/NREL/EnergyPlus/wiki/Debugging-Tips)

**Comment by mjwitte** [comment]
@JasonGlazer Is this ready for review?

**Comment by JasonGlazer** [comment]
Almost, it looks like some unit test is failing.

**Comment by Myoldmopar** [comment]
@JasonGlazer I fixed the unit test failure.  The problem was that the fluid index was totally uninitialized, and it happened to be a positive value.  Inside GetDensityGlycol, it checks if the passed in index is `>0`, and if so, it uses that as the index.  I changed it to a zero initialization, and now the GetDensityGlycol function properly looks up an index by name.  At the same time I made the name upper case...not sure if that was needed, but it didn't hurt.

**Comment by JasonGlazer** [comment]
@Myoldmopar Thanks!

**Comment by JasonGlazer** [comment]
Looks like this one is passing all the CI tests now and is ready to merge.

**Pull request merged**